paho/message_ids.go

```diff
@@ -75,14 +78,14 @@ func (m *MIDs) Request(c *CPContext) (uint16, error) {
 func (m *MIDs) Get(i uint16) *CPContext {
 	m.Lock()
 	defer m.Unlock()
-	return m.index[i]
+	return m.index[i-1]
```

Comment:
It may be worth adding a comment in the `struct` definition ```index []*CPContext // index is messageid-1``` (just to avoid future misunderstandings)

Reply:
Thanks you :)
Current original (master) inplemention is index == messageid, so I also think the comment is worth.